src/include/migraphx/op/slice.hpp

```diff
@@ -144,11 +146,12 @@ struct slice
             auto sliced_length =
                 fix_index(old_lens, axis, ends[i]) - fix_index(old_lens, axis, starts[i]);
             // A Numpy indexing convention: a slice size larger than the actual dimension
-            // is legal and is clipped to the axis size
+            // is legal and the "ends" value is clipped to the axis size
             new_lens[axis] = std::min(new_lens[axis], sliced_length);
             if(input_shape.dynamic())
             {
-                new_mins[axis] = std::min(new_lens[axis], new_mins[axis]);
+                auto sliced_min_length = fix_index(new_mins, axis, ends[i]) - fix_index(new_mins, axis, starts[i]);
+                new_mins[axis] = std::min(sliced_min_length, new_mins[axis]);
```

Comment:
I would find it better if we used `old_mins`, `old_maxs` etc. as names. 
Here we are using `new_mins` to calculuate and values to `new_mins` which defeats the `new` meaning

Reply:
I'd like to leave this as is.  I think it's already about as clear as any alternative I can think of.  The new_mins array is copied from the inputs, then one or a few of its values are updated.